### ISSUE TITLE
No 'tab' possible on 'else:' condition

### ISSUE THREAD

**Issue by csteiiin** [GDQuest/emacs-gdscript-mode#52]
**I'm submitting a...**

- [x] Bug report.
- [ ] Feature request.

## Bug report ##

**What is the current behavior?**
It is not possible to tab the "else" case inside a funkction. 

**What is the expected behavior?**
I should be able to tab 'else:' to be on the same level then the 'if' condition.

**Tell us the steps to reproduce the bug, and if possible share a minimal demo of the problem.** 
A project with only an Area2D (Player) an AnimatedSprite and a CollisionShape2D
With a Mark on 'e' of 'else:' and pressing tab as well as `C-x h tab` the 'else:' does not change the position. 
That way I get a build error in Godot.
```
extends Area2D

# movement speed of the player
export var speed = 400

# size of the game window
var screen_size

func _ready():
	# load screen size
	screen_size = get_viewport_rect().size

# delta = frame length
func _process(delta):
	# player's movement vector
	var velocity = Vector2()

	# set movement direction depending on pressed button
	if Input.is_action_pressed("ui_right"):
		velocity.x -= 1
	if Input.is_action_pressed("ui_left"):
		velocity.x += 1
	if Input.is_action_pressed("ui_up"):
		velocity.y -= 1
	if Input.is_action_pressed("ui_down"):
		velocity.y += 1

	if velocity.lenght() > 0:
		velocity = velocity.normalized() * speed
		$AnimatedSprite.play()
else:
	$AnimatedSprite.stop()
	
position += velocity * delta
position.x = clamp(position.x, 0, screen_size.x)
position.x = clamp(position.y, 0, screnn_size.y)
```

## Other ##

**Other information** 
The steps are done processed by [this guide.](https://docs.godotengine.org/de/latest/getting_started/step_by_step/your_first_game.html)
![Screenshot_2020-04-28_23-49-43](https://user-images.githubusercontent.com/38066436/80541202-0684c980-89ab-11ea-94de-dabb01ba3735.png)

I use the latest official emacs on a ubuntu 20.04 Studio machine. 
In godot i have set the external editor to:
- Exec Path: emacsclient
- Exec Flags: -c -a '' {file}

That works totally fine.
I can also shift the "if ..." condition totally free so that works fine.

**Comment by NathanLovato** [comment]
It seems the indentation is generally off for the else keyword in some cases. It shouldn't reset to 0.

**Comment by csteiiin** [comment]
I'am just starting with godot but do use emacs on a daily bases.
My lisp knowledge is very low but I want to help as much as possible to be able to use my favourite editor/tool together with godot.

**Comment by csteiiin** [comment]
Debuggin into it with this [Guide](http://blog.binchen.org/posts/my-work-flow-to-debug-emacs-plugin.html) brings me to the follwing situation.

![Screenshot_2020-05-01_16-57-47](https://user-images.githubusercontent.com/38066436/80815031-3c08fd00-8bcd-11ea-91c7-62d553a31887.png)

At first I was doing a `M-x gdscript-indent-shift-right`to bring the `else` to the right possition. With  `edebug-defun` while pressing `tab`it jumps in to the `gdscript-indent-line` function. with `e` i looked for the `previous` expression that resulted in a `nil`. 

Stepping through it the first `(let ...)` part had a `t` in both conditions.
Stepping through `(gdscript-indent-calculate-indentation previous)` resulted in `nil` (That is my suspission of the error spot) everything in there resulted in `0`or `nil` 

`(when follow-indentation-p ...)` as well was all resulting in `nil`

I hope that helps.
Actually I am not capable of going deeper I think (at least not very efficient). I will try but I hope you can also find time for it.

**Comment by NathanLovato** [comment]
Thanks much for the details. At the moment I'm really busy, I don't have time for this in the next few days, but I'll get to it if nobody beats me to it.

**Comment by csteiiin** [comment]
Same behavior for `elif`

**Comment by NathanLovato** [comment]
This was fixed by @VlachJosef in #55. Closing now.
